pkg/logs/tail.go

```diff
@@ -201,6 +201,13 @@ func (c *TailCommand) tail(out io.Writer) {
 		// Check that our request was successful. If the server is
 		// having trouble, retry after waiting for some time.
 		if resp.StatusCode != http.StatusOK {
+			// If the response was a 404, the from time was
+			// not valid, give them an error stating this and exit.
+			if resp.StatusCode == http.StatusNotFound &&
+				c.cfg.from != 0 {
```

Comment:
Sorry if I'm missing context, how does `c.cfg.from != 0` mean the from date is invalid?
I fear this would catch and mislabel any 404 error?

Reply:
@jaredmorrow are you able to provide any extra context? I understand the point @ejthurgo is making but I was under the impression this logic was based on a specific behaviour inherent to your API (e.g. in the sense that the API would only return a 404 when an incorrect flag value was provided). 

If that's not the case and the API is expected to return a 404 in other scenarios, then yes this could end up be an accidental match where the user provides a valid flag value but something else in their request has caused the API to return a 404.

Reply:
The combination of the 404 & the from field being set is the key here.  If from is set and it is out of range, the 404 returned would be due to the range being too far out from Now.  There are currently no other 404s sent by the server, but just in case I added the `from != 0` to be specific about the case.

Reply:
Surprising the server doesn't give any specific detail on why it failed? Relying on the fact it **currently** only sends a 404 for this circumstance seems awfully risky.
But thanks for the context.